src/sensors/mod.rs

```diff
@@ -0,0 +1,93 @@
+//! Module defining common sensor functionality.
+
+mod humidity;
+mod people_now_present;
+mod power_consumption;
+mod temperature;
+
+pub use humidity::{HumiditySensor, HumiditySensorTemplate};
+pub use people_now_present::{PeopleNowPresentSensor, PeopleNowPresentSensorTemplate};
+pub use power_consumption::{PowerConsumptionSensor, PowerConsumptionSensorTemplate};
+pub use temperature::{TemperatureSensor, TemperatureSensorTemplate};
+
+use log::warn;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+/// Common information describing any sensor.
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct SensorMetadata {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub location: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+}
+
+/// Common information describing any sensor which requires a specified location.
+#[derive(Serialize, Deserialize, Default, Debug, Clone, PartialEq)]
+pub struct LocalisedSensorMetadata {
```

Comment:
Nothing against the British spelling, but for identifiers in software I'd generally prefer the American forms ("localized"). This is also consistent with Serde ("serialize", "deserialize").

In this case, I'd even question whether "localized" is a good prefix. When I hear the term, I first think about localization in the sense of internationalization/translation 🙂 Maybe `SensorMetdataWithLocation`?

Reply:
Heh, yes. Was never really happy with that name myself.